### PR TITLE
Fix  subscription not persisted in  Solace broker without any Consumer Key/Secret

### PR DESCRIPTION
**Pull request by Sarangan0219**
### Purpose

- Fixes https://github.com/wso2/product-apim/issues/12691
- Fixes https://github.com/wso2/product-apim/issues/12695

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.solace/src/main/java/org/wso2/carbon/apimgt/solace/notifiers/SolaceKeyGenNotifier.java [hunk 1]
```diff
@@ -97,35 +105,66 @@ private void syncSolaceApplicationClientId(ApplicationRegistrationEvent event) t
                     if (gatewayEnvironments.containsKey(deployment.getDeployment())) {
                         if (SolaceConstants.SOLACE_ENVIRONMENT.equalsIgnoreCase(gatewayEnvironments.get(deployment.
                                 getDeployment()).getProvider())) {
+                            APIProvider apiProvider = APIManagerFactory.getInstance().getAPIProvider(CarbonContext.
```
Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.solace/src/main/java/org/wso2/carbon/apimgt/solace/notifiers/SolaceKeyGenNotifier.java:108]: can't we internally retrieve this ?

File: components/apimgt/org.wso2.carbon.apimgt.solace/src/main/java/org/wso2/carbon/apimgt/solace/notifiers/SolaceKeyGenNotifier.java [hunk 2]
```diff
@@ -97,35 +105,66 @@ private void syncSolaceApplicationClientId(ApplicationRegistrationEvent event) t
...
                             consumerSecret = key.getConsumerSecret();
+                            application.addKey(key);
+                        }
+                    }
+                    // Send only the production keys to Solace broker.
+                    if (application.getKeys().isEmpty()) {
+                        return;
+                    }
+                    SolaceAdminApis solaceAdminApis = SolaceNotifierUtils.getSolaceAdminApis();
+                    CloseableHttpResponse isApplicationExistsResponse = solaceAdminApis.applicationGet(
```
Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.solace/src/main/java/org/wso2/carbon/apimgt/solace/notifiers/SolaceKeyGenNotifier.java:138]: add try-resource

**Review by tharindu1st — CHANGES_REQUESTED** [review]

File: components/apimgt/org.wso2.carbon.apimgt.solace/src/main/java/org/wso2/carbon/apimgt/solace/notifiers/SolaceKeyGenNotifier.java (hunk 1 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.solace/src/main/java/org/wso2/carbon/apimgt/solace/notifiers/SolaceKeyGenNotifier.java:108]: We need an API object to be passed to create an application Solace broker.
That is retrieved from the subscriprtionEvent in subscription flow. In key generation flow, we don't have that object in the event and we need to access the DAO layer.

File: components/apimgt/org.wso2.carbon.apimgt.solace/src/main/java/org/wso2/carbon/apimgt/solace/notifiers/SolaceKeyGenNotifier.java (hunk 2 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.solace/src/main/java/org/wso2/carbon/apimgt/solace/notifiers/SolaceKeyGenNotifier.java]: Fixed https://github.com/wso2/carbon-apimgt/pull/11173/commits/6b68db143bdcb3bb3d228ba654e0c4d500188c44

File: components/apimgt/org.wso2.carbon.apimgt.solace/src/main/java/org/wso2/carbon/apimgt/solace/notifiers/SolaceKeyGenNotifier.java (hunk 1 above)
Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.solace/src/main/java/org/wso2/carbon/apimgt/solace/notifiers/SolaceKeyGenNotifier.java]: Fixed https://github.com/wso2/carbon-apimgt/pull/11173/commits/b4731c797a76dce959fc2727b1530e0660ef2b54

**Review by tharindu1st — APPROVED** [review]

**Pull request merged**